packages/fluentui/react-component-event-listener/src/EventListenerWithComputedTarget.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+import useEventListener from './useEventListener';
+import { EventListenerOptions, EventTypes, Target } from './types';
+
+type EventListenerWithComputedTargetProps<T extends EventTypes> = Omit<EventListenerOptions<T>, 'target' | 'targetRef'> & {
+  resolveTarget: () => Target;
+};
+
+function EventListenerWithComputedTarget<T extends EventTypes>(props: EventListenerWithComputedTargetProps<T>) {
+  const { resolveTarget, ...rest } = props;
+  const targetRef = React.useRef<Target | null>(null);
+
+  React.useEffect(() => {
+    targetRef.current = resolveTarget();
+  }, [resolveTarget]);
+  useEventListener({ ...rest, target: undefined, targetRef });
```

Comment:
how come target is set to undefined here?

Reply:
Oops, it should not be there